test/bats/test.bats

```diff
@@ -12,190 +12,166 @@ teardown() {
-
-  run wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod -l control-plane=controller-manager"
-  assert_success
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod -l control-plane=controller-manager"
 }
 
 @test "gatekeeper-audit is running" {
-  run wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod -l control-plane=audit-controller"
-  assert_success
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod -l control-plane=audit-controller"
 }
 
 @test "namespace label webhook is serving" {
-
   cert=$(mktemp)
   CLEAN_CMD="${CLEAN_CMD}; rm ${cert}"
-  run wait_for_process $WAIT_TIME $SLEEP_TIME "get_ca_cert ${cert}"
-  assert_success
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "get_ca_cert ${cert}"
 
   kubectl run temp --generator=run-pod/v1  --image=tutum/curl -- tail -f /dev/null
   kubectl wait --for=condition=Ready --timeout=60s pod temp
   kubectl cp ${cert} temp:/cacert
 
-  run wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl exec -it temp -- curl -f --cacert /cacert --connect-timeout 1 --max-time 2  https://gatekeeper-webhook-service.gatekeeper-system.svc:443/v1/admitlabel"
-  assert_success
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl exec -it temp -- curl -f --cacert /cacert --connect-timeout 1 --max-time 2  https://gatekeeper-webhook-service.gatekeeper-system.svc:443/v1/admitlabel"
   kubectl delete pod temp
 }
 
-
 @test "constrainttemplates crd is established" {
-  wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/constrainttemplates.templates.gatekeeper.sh"
-
-  run kubectl -n gatekeeper-system get crd/constrainttemplates.templates.gatekeeper.sh
-  assert_success
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/constrainttemplates.templates.gatekeeper.sh"
+  kubectl -n gatekeeper-system get crd/constrainttemplates.templates.gatekeeper.sh
 }
 
 @test "waiting for validating webhook" {
-  wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration"
-
-  run kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration
-  assert_success
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration"
+  kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration
 }
 
 @test "applying sync config" {
-  run kubectl apply -f ${BATS_TESTS_DIR}/sync.yaml
-  assert_success
```

Comment:
why are we removing assert?

Reply:
run/assert are generally not recommended for simple return value tests, as `set -e` is in place:

https://github.com/bats-core/bats-core#testing

Reply:
There was a discussion about this in the thread with mmirecki above.